### PR TITLE
Update dotnet-core 3.1 to 3.1.10

### PR DESCRIPTION
**Pull request by liushuyu**
Topic Description
-----------------

Update `dotnet` 3.1 series to 3.1.10

Package(s) Affected
-------------------

- extra-libs/dotnet-host
- extra-libs/dotnet-host-fxr
- extra-libs/dotnet-runtime-3.1
- extra-devel/aspnetcore-runtime-3.1
- extra-devel/dotnet-sdk-3.1.4xx
- extra-devel/dotnet-sdk-3.1.1xx
- extra-devel/dotnet-sdk-3.1

Security Update?
----------------

Yes, `#PENDING`

Build Order
-----------

```
extra-libs/dotnet-host
extra-libs/dotnet-host-fxr
extra-libs/dotnet-runtime-3.1
extra-devel/aspnetcore-runtime-3.1
extra-devel/dotnet-sdk-3.1.4xx
extra-devel/dotnet-sdk-3.1.1xx
extra-devel/dotnet-sdk-3.1
```

Architectural Progress
----------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

Secondary Architectural Progress
--------------------------------

N/A

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

Post-Merge Secondary Architectural Progress
-------------------------------------------

N/A

---
This change is [Reviewable](https://reviewable.io/reviews/aosc-dev/aosc-os-abbs/2576)

### CODE REVIEW

File: extra-libs/dotnet-host/autobuild/build [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
+abinfo "Deploying files ..."
+mkdir -pv "$PKGDIR"/usr/bin
+mkdir -pv "$PKGDIR"/usr/lib/dotnet
+mkdir -pv "$PKGDIR"/usr/share/doc/dotnet
+install -Dvm755 "$SRCDIR"/dotnet "$PKGDIR"/usr/lib/dotnet
+install -Dvm644 "$SRCDIR"/LICENSE.txt "$PKGDIR"/usr/share/doc/dotnet
+install -Dvm644 "$SRCDIR"/ThirdPartyNotices.txt "$PKGDIR"/usr/share/doc/dotnet
 # cp dotnet.1.gz "$PKGDIR"/usr/share/man/man1
-ln -s /usr/lib/dotnet/dotnet "$PKGDIR"/usr/bin/dotnet
+ln -sv /usr/lib/dotnet/dotnet "$PKGDIR"/usr/bin/dotnet
```
Comment (MingcongBai) [inline extra-libs/dotnet-host/autobuild/build]: It's probably more reasonable to use a relative symlink target here, i.e. `../lib/dotnet/dotnet`.

File: extra-libs/dotnet-host/autobuild/build [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
-cp "$SRCDIR"/LICENSE.txt "$PKGDIR"/usr/share/doc/dotnet
-cp "$SRCDIR"/ThirdPartyNotices.txt "$PKGDIR"/usr/share/doc/dotnet
+abinfo "Deploying files ..."
+mkdir -pv "$PKGDIR"/usr/bin
+mkdir -pv "$PKGDIR"/usr/lib/dotnet
+mkdir -pv "$PKGDIR"/usr/share/doc/dotnet
+install -Dvm755 "$SRCDIR"/dotnet "$PKGDIR"/usr/lib/dotnet
+install -Dvm644 "$SRCDIR"/LICENSE.txt "$PKGDIR"/usr/share/doc/dotnet
+install -Dvm644 "$SRCDIR"/ThirdPartyNotices.txt "$PKGDIR"/usr/share/doc/dotnet
 # cp dotnet.1.gz "$PKGDIR"/usr/share/man/man1
```
Comment (MingcongBai) [inline extra-libs/dotnet-host/autobuild/build]: Drop this line if not needed.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Comment by MingcongBai** [comment]
@liushuyu Any updates?

**Comment by liushuyu** [comment]
Superseded by #2717

**Pull request closed without merging**